numba/core/typing/new_builtins.py

```diff
@@ -0,0 +1,1178 @@
+        """
+        assert not kws
+
+        if not args:
+            return
+        if len(args) == 1:
+            # max(arg) only supported if arg is an iterable
+            if isinstance(args[0], types.BaseTuple):
+                tys = list(args[0])
+                if not tys:
+                    raise TypeError("%s() argument is an empty tuple"
+                                    % (self.key.__name__,))
+            else:
+                return
+        else:
+            # max(*args)
+            tys = args
+        retty = self._unify_minmax(tys)
+        if retty is not None:
+            return signature(retty, *args)
+
+
+@infer_global(max)
+class Max(MinMaxBase):
+    pass
+
+
+@infer_global(min)
+class Min(MinMaxBase):
+    pass
+
+
+@infer_global(round)
+class Round(ConcreteTemplate):
+    cases = [
+        signature(types.np_intp, types.np_float32),
+        signature(types.np_int64, types.np_float64),
+        signature(types.np_float32, types.np_float32, types.np_intp),
+        signature(types.np_float64, types.np_float64, types.np_intp),
+    ]
+
+
+#------------------------------------------------------------------------------
+
+
+@infer_global(bool)
+class Bool(AbstractTemplate):
+
+    def generic(self, args, kws):
+        assert not kws
+        [arg] = args
+        if isinstance(arg, (types.Boolean, types.Number)):
+            return signature(types.np_bool_, arg)
+        # XXX typing for bool cannot be polymorphic because of the
+        # types.Function thing, so we redirect to the operator.truth
+        # intrinsic.
+        return self.context.resolve_function_type(operator.truth, args, kws)
+
+
+@infer_global(int)
+class Int(AbstractTemplate):
+
+    def generic(self, args, kws):
+        if kws:
+            raise errors.NumbaAssertionError('kws not supported')
+
+        [arg] = args
+
+        if isinstance(arg, types.Integer):
+            return signature(arg, arg)
+        if isinstance(arg, (types.Float, types.Boolean)):
+            return signature(types.np_intp, arg)
+        if isinstance(arg, types.NPDatetime):
+            if arg.unit == 'ns':
+                return signature(types.np_int64, arg)
+            else:
+                raise errors.NumbaTypeError(f"Only datetime64[ns] can be converted, but got datetime64[{arg.unit}]")
+        if isinstance(arg, types.NPTimedelta):
+            return signature(types.np_int64, arg)
+
+
+@infer_global(float)
+class Float(AbstractTemplate):
+
+    def generic(self, args, kws):
+        assert not kws
+
+        [arg] = args
+
+        if arg not in types.number_domain:
+            raise errors.NumbaTypeError("float() only support for numbers")
+
+        if arg in types.np_complex_domain:
+            raise errors.NumbaTypeError("float() does not support complex")
+
+        if arg in types.integer_domain:
+            return signature(types.np_float64, arg)
+
+        elif arg in types.np_real_domain:
+            return signature(arg, arg)
```

Comment:
The `float` constructor should have signature which have a hard-coded return type of `py_float64`?